tests/update/patch_before_migration.pm

```diff
@@ -126,6 +126,32 @@ sub install_patterns {
     }
 }
 
+sub sle_register {
+    my ($action) = @_;
+    # Register sle before update
+    if ($action eq 'register') {
+        if (sle_version_at_least('12')) {
+            yast_scc_registration();
+        }
+        else {
+            my $reg_code = get_required_var("NCC_REGCODE");
+            my $reg_mail = get_required_var("NCC_MAIL");
```

Comment:
Is mail address mandatory for ncc registration? If not, we can just keep NCC_REGCODE.

Reply:
Yes, it is mandatory.

Reply:
I'm not sure about this, because seems scc registration only require SCC_RGCODE with alpha keys.
It's fine if you have confirmed that ncc requires both mail and regcode even with testing keys (like alpha keys for scc).

Reply:
From suse_rigster help, NCC_MAIL is mandatory.
The regitser commad should be like this
suse_register -n -a email=compay@example -a regcode-sles=03271221qwetnsk3429934

Reply:
Thanks for the confirmation.